server/controller/user.js

```diff
@@ -0,0 +1,116 @@
+import db from '../models/index';
+import { validateSignup, validateLogin, comparePwd } from '../validators/validator.js';
+import { sendValidationError, serverError, sendSuccess, sendFail } from '../reply/reply';
+import { generateToken } from '../auth/auth';
+const Users = db.Users;
+const log = console.log;
+
+
+export const validateSignupData = (req, res, next) => {
+  const signupData = {
+    username: req.body.username,
+    password: req.body.password,
+    email: req.body.email,
+  };
+  const validate = validateSignup(signupData);
+  if (validate.valid) {
+    req.body = signupData;
+    next();
```

Comment:
why do you have to use next()?

Reply:
an calling the next middleware function.